x-pack/plugins/security_solution/public/management/pages/policy/view/trusted_apps/flyout/policy_trusted_apps_flyout.tsx

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useState, useCallback, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+import { isEmpty } from 'lodash/fp';
+import {
+  EuiButton,
+  EuiTitle,
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiFlyoutBody,
+  EuiSpacer,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiEmptyPrompt,
+} from '@elastic/eui';
+import {
+  policyDetails,
+  getCurrentArtifactsLocation,
+  getAvailableArtifactsList,
+  getAvailableArtifactsListIsLoading,
+  getUpdateArtifactsIsLoading,
+  getUpdateArtifactsLoaded,
+  getAvailableArtifactsListExist,
+  getAvailableArtifactsListExistIsLoading,
+} from '../../../store/policy_details/selectors';
+import {
+  usePolicyDetailsNavigateCallback,
+  usePolicyDetailsSelector,
+  usePolicyTrustedAppsNotification,
+} from '../../policy_hooks';
+import { PolicyArtifactsList } from '../../artifacts/list';
+import { SearchExceptions } from '../../../../../components/search_exceptions';
+
+export const PolicyTrustedAppsFlyout = React.memo(() => {
+  usePolicyTrustedAppsNotification();
+  const dispatch = useDispatch();
+  const [selectedArtifactIds, setSelectedArtifactIds] = useState<string[]>([]);
+  const location = usePolicyDetailsSelector(getCurrentArtifactsLocation);
+  const policyItem = usePolicyDetailsSelector(policyDetails);
+  const availableArtifactsList = usePolicyDetailsSelector(getAvailableArtifactsList);
+  const isAvailableArtifactsListLoading = usePolicyDetailsSelector(
+    getAvailableArtifactsListIsLoading
+  );
+  const isUpdateArtifactsLoading = usePolicyDetailsSelector(getUpdateArtifactsIsLoading);
+  const isUpdateArtifactsLoaded = usePolicyDetailsSelector(getUpdateArtifactsLoaded);
+  const isAvailableArtifactsListExist = usePolicyDetailsSelector(getAvailableArtifactsListExist);
+  const isAvailableArtifactsListExistLoading = usePolicyDetailsSelector(
+    getAvailableArtifactsListExistIsLoading
+  );
+
+  const policyName = policyItem?.name ?? '';
+
+  const handleListFlyoutClose = usePolicyDetailsNavigateCallback(() => ({
+    show: undefined,
+  }));
+
+  useEffect(() => {
+    if (isUpdateArtifactsLoaded) {
+      handleListFlyoutClose();
+      dispatch({
+        type: 'policyArtifactsUpdateTrustedAppsChanged',
+        payload: { type: 'UninitialisedResourceState' },
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
same here. We should really avoid this. 

Looking at this, feels like this belongs in the middleware, no?